src/app/services/pannellum.service.ts

```diff
@@ -1,502 +1,555 @@
+  * @param hotSpotDiv div del hotspot
+  * @param args Argumentos custom del hotspot
+  */
+  public hotspot(hotSpotDiv, args) {
+
+    console.log(`Cargando Hotspot ${args.title}`);
+
+    // Custom class
+    hotSpotDiv.classList.add('custom-tooltip');
+
+    // Custom ID
+    hotSpotDiv.id = args.id;
+
+
+    // Se crea el evento para abrir el modal
+    if (args.modal) {
+
+      if ( args.modal.type === "db"){
+        this.apiServive.getArtefact(args.modal.id).subscribe(
+          data =>{
+
+            let modalData = {
+              'title': data.result[0].artifactLabel.value || null,
+              'description': data.result[0].note.value || null,
+              'imagen': {
+                'src': args.modal.imagen.src,
+                'alt': args.modal.imagen.alt,
+                'width': args.modal.imagen.width,
+                'height': args.modal.imagen.height,
+              }
+            }
+
+            let modal = document.getElementById(args.id)
+            modal.onclick = () => this.openModal(modalData)
+
+          }
+        );
+      }
+      else{
+        let modal = document.getElementById(args.id)
+        modal.onclick = () => this.openModal(args.modal)
+      }
+
+    }
+
+    // Create span element to tooltip
+    var span = document.createElement('span');
+    span.innerHTML = args.title;
+    hotSpotDiv.appendChild(span);
+    span.style.width = span.scrollWidth - 20 + 'px';
+    span.style.marginLeft = (- (span.scrollWidth - hotSpotDiv.offsetWidth) / 2 + 24) + 'px';
+    span.style.marginTop = -span.scrollHeight - 12 + 'px';
+
+    span.classList.add('custom-tooltip-span');
+
+    // Se crea el evento para abrir el modal
+    if (args.modal && args.modal.title) {
+      let modal = document.getElementById(args.id)
+      modal.onclick = () => this.openModal(args.modal)
+    }
+
+    // Custom icon
+    if (args.customIcon && args.customIcon.src) {
+      // Quitamos la clase de pannellum para quitar el icono por defecto
+      hotSpotDiv.classList.remove('pnlm-hotspot')
+
+      let width = args.customIcon.width || "50";
+      let height = args.customIcon.height || "50";
+
+      $(`#${args.id}`).append(
+        `<img src="${args.customIcon.src}" alt="${args.customIcon.alt}" width="${width}" height="${height}">`
+      );
+    }
+
+  }
+
+  /**
+   * OutputJson
+   *
+   * Genera un json a patir de las scenas almacenadas
+   */
+  public OutputJson() {
+
+
+
+  }
+
+
+  /**
+   * setScene
+   * Establece la escena a mostrar
+   * @param sceneId id de la escena
+   */
+  setScene = (sceneId: string): void => {
+    if (this.pannellumViewer) this.pannellumViewer.loadScene(sceneId);
+  }
+
+  public getAllHotspots() {
+
+    let ListHotspots = [];
+    if ( this.sceneJson ){
+
+      for (let i in this.scenes) {
+       // console.log(index2); // prints elements: 0, 1, 2, 3
+        for (let j of this.sceneJson[this.scenes[i]]['hotSpots']){
+          ListHotspots.push(j)
+        }

```

Comment:
```suggestion
    hotspots = [...hotspots, ...this.sceneJson[this.scenes[i]]];
```
Notese que sugiero cambiar el nombre `ListHotspots` a `hotspots`. Las variables no deberian empezar en mayuscula